### PR TITLE
Use chromium_src override for ethash override

### PR DESCRIPTION
**Pull request by bbondy**
It's better to use a chromium_src override because source files are touched automatically.

Resolves https://github.com/brave/brave-browser/issues/15798

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: chromium_src/third_party/ethash/src/lib/keccak/keccak.c [hunk 1]
```diff
@@ -7,4 +7,4 @@
 #undef __x86_64__
 #endif
```
Comment (goodov) [inline chromium_src/third_party/ethash/src/lib/keccak/keccak.c:9]: should be ```chromium_src/```**brave**```/third_party/ethash/src/lib/keccak/keccak.c```

and there is a typo in a comment above: "needd on macOS"

File: chromium_src/third_party/ethash/src/lib/keccak/keccak.c [hunk 2]
```diff
@@ -7,4 +7,4 @@
...
 
-#include "src/lib/keccak/keccak.c"
+#include "../../../third_party/ethash/src/lib/keccak/keccak.c"
```
Comment (goodov) [inline chromium_src/third_party/ethash/src/lib/keccak/keccak.c:10]: it seems like we should have 7 parents here (```../```) to have root at ```./src/brave```
```
brave/chromium_src/brave/third_party/ethash/src/lib/keccak
    7            6     5           4      3   2   1
```

```#include "../../../../../../../third_party/ethash/src/lib/keccak/keccak.c"```

Comment (bbondy) [reply chromium_src/third_party/ethash/src/lib/keccak/keccak.c]: Yep, I would have noticed from CI on Windows but I'm working from a mac. Thanks. Fixed in 8e6f04b3ec681d9478bdf1c2e785a43937730458

File: chromium_src/third_party/ethash/src/lib/keccak/keccak.c (hunk 1 above)
Comment (bbondy) [reply chromium_src/third_party/ethash/src/lib/keccak/keccak.c]: FixedFixed in 444c3346445a8615688dab4532d41889d99c648e

**Review by goodov — APPROVED** [review]
typo is still there

**Comment by bbondy** [comment]
> typo is still there

Fixed in a372638cace6ccfd7489334b9af115864cd7bf1b

**Pull request merged**